src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -28,14 +29,147 @@ static ucs_status_t uct_rdmacm_cm_query(uct_cm_h cm, uct_cm_attr_t *cm_attr)
     return UCS_OK;
 }
 
-static void uct_rdmacm_cm_event_handler(int fd, void *arg){}
 
+static void
+uct_rdmacm_cm_process_event(uct_rdmacm_cm_t *cm, struct rdma_cm_event *event)
+{
+    struct sockaddr            *remote_addr = rdma_get_peer_addr(event->id);
+    uint8_t                    ack_event = 1;
+    struct rdma_conn_param     conn_param;
+    uct_rdmacm_priv_data_hdr_t *hdr;
+    uct_rdmacm_cm_ep_t         *cep;
+    char                       dev_name[UCT_DEVICE_NAME_MAX];
+    char                       ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    ucs_status_t               status;
+    ssize_t                    priv_data_ret;
+
+    ucs_trace("rdmacm event (fd=%d cm_id %p): %s. Peer: %s.",
+              cm->ev_ch->fd, event->id, rdma_event_str(event->event),
+              ucs_sockaddr_str(remote_addr, ip_port_str, UCS_SOCKADDR_STRING_LEN));
+
+    /* The following applies for rdma_cm_id of type RDMA_PS_TCP only */
```

Comment:
IMHO it's an overkill to do on every event, since to do that need to retrieve the ip and port address, call rdma_getaddrinfo() and read the result to get to port space. not to mention the calloc and free (rdma_freeaddrinfo()) that it does.

Reply:
let's take from rdmacm id -> ps